### ISSUE TITLE
bun install realm package missing file

### ISSUE THREAD

**Issue by kwanLeeFrmVi** [oven-sh/bun#6098]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Macos Darwin 22.5.0 arm64 arm

### reproduce step
0# 
install realm package in a nestjs project

1#
run `bun install realm`
output: 
```
bun add v1.0.0 (822a00c4)
 installed realm@12.2.0
[325.00ms] done
```
2#
try to call realm 
```
// app.module.ts
import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import * as Realm from 'realm';

@Module({
  imports: [],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {
  constructor() {
    const app = new Realm.App({ id: '<alias-mongodb-app-id>' });
    const credentials = Realm.Credentials.anonymous();
    try {
      app.logIn(credentials).then(console.log);
    } catch (err) {
      console.error('Failed to log in', err);
    }
  }
}

```

3# 
run `bun run start:dev`
output
```
[4:32:25 PM] Starting compilation in watch mode...

[4:32:26 PM] Found 0 errors. Watching for file changes.

Error: Cannot find module '/Users/<username>/Documents/regenesis-api-v2/node_modules/realm/realm-constants.json'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1096:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1089:15)
    at trySelf (node:internal/modules/cjs/loader:541:12)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1052:24)
    at Function.Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/Users/quanleminh/Documents/regenesis-api-v2/node_modules/realm/dist/bundle.node.js:10:27)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
```

### my solution is:
install package by using `yarn add realm` or `yarn` (if package already in package.json)
before run app by `bun run start:dev`

### Additional information

I think bun install miss node_modules/realm/realm-constants.json file 
which contain
``` 
{"REALM_ANONYMIZED_BUNDLE_ID":"NqR4aRH75ptA< a random string >L1gPKFH++wCkUmryWs="}
```

**Comment by Electroid** [comment]
Duplicate of #4959

**Comment by Electroid** [comment]
This is because the `postinstall` script is not running, see #4959 for more details and a temporary workaround.

**Comment by ansarizafar** [comment]
The workaround is not working for realm package
```json
{
  "name": "realmdb",
  "module": "index.js",
  "devDependencies": {
    "bun-types": "latest"
  },
  "peerDependencies": {
    "typescript": "^5.0.0"
  },
 
  "type": "module",
  "dependencies": {
    "realm": "^12.2.0"
  },
  "trustedDependencies": ["realm"]
}
```
Here is the error message
zafaransari@Latitude-E7470:~/projects/bun/realmdb$ bun .


error: Cannot find module "realm/realm-constants.json" from "/home/zafaransari/projects/bun/realmdb/node_modules/realm/dist/bundle.node.js"

**Comment by kwanLeeFrmVi** [comment]
<del>I solve it by add `wireit` to trustedDependencies</del>
workaround is not working
I have to install the package by npm 
before run by bun

**Comment by Jarred-Sumner** [comment]
Fixed by @dylan-conway and @paperdave in #7132

This will be part of the Bun v1.0.17 release, which ships in a few hours from the time of writing.
